### PR TITLE
feat: copy lightning address button in receive screen

### PR DESCRIPTION
**Pull request by pavanjoshi914**
### Describe the changes you have made in this PR

copy the lightning address button in receiving screen

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

https://github.com/getAlby/lightning-browser-extension/assets/55848322/e74d600f-dce3-4381-bfe9-0186a4a63fd8

### Checklist

- [X] Self-review of changed code
- [X] Manual testing
- [X] Added automated tests where applicable
- [X] Update Docs & Guides
- For UI-related changes
- [X] Darkmode
- [X] Responsive layout

### CODE REVIEW

File: src/app/screens/Receive/index.tsx [hunk 1]
```diff
@@ -291,6 +304,34 @@ function Receive() {
                   <div className="flex-grow border-t  border-gray-300 dark:border-gray-700"></div>
                 </div>
 
+                <div className="mb-4">
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/screens/Receive/index.tsx:308]: https://github.com/getAlby/lightning-browser-extension/pull/2565/files#diff-8efedef6f0732512da2fd92510e9dead5a4a79577226f1cf3bd68a1b900647edL283

we already have a check here! it was added in receive via bitcoin addr button Pr

[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/screens/Receive/index.tsx:308]: image

File: src/app/screens/Receive/index.tsx [hunk 2]
```diff
@@ -41,19 +41,23 @@ function Receive() {
...
+  const [copyLightningAddressLabel, setCopyLightningAddressLabel] = useState(
+    t("actions.copy_lightning_address") as string
+  );
+  const [lightningAddress, setLightningAddress] = useState("");
   const [paid, setPaid] = useState(false);
   const [pollingForPayment, setPollingForPayment] = useState(false);
   const mounted = useRef(false);
   const isAlbyUser =
-    isAlbyOAuthAccount(auth.account?.alias) ||
+    isAlbyOAuthAccount(auth.account?.connectorType) ||
```
Comment (rolznz) [inline src/app/screens/Receive/index.tsx:60]: This is already fixed in master, why is it not in this diff?

File: src/app/screens/Receive/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (reneaaron) [reply src/app/screens/Receive/index.tsx:308]: You are right, I missed that in the diff view. Sorry :sweat_smile:

File: src/app/screens/Receive/index.tsx [hunk 3]
```diff
@@ -291,6 +305,32 @@ function Receive() {
...
                 </div>
 
+                <div className="mb-4">
+                  <Button
+                    type="button"
+                    label={copyLightningAddressLabel}
+                    fullWidth
+                    onClick={async () => {
+                      try {
+                        if (lightningAddress) {
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/screens/Receive/index.tsx:315]: else part was added before if you see the video! but it was removed as every alby user now has a lightning address cc @reneaaron

[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/screens/Receive/index.tsx:315]: disabling button till account info is loaded

File: src/app/screens/Receive/index.tsx [hunk 4]
```diff
@@ -140,6 +144,16 @@ function Receive() {
     createInvoice();
   }
 
+  async function getLightningAddress() {
+    const response = await api.getAccountInfo();
```
[the remark this answers is not in the document]
Comment (pavanjoshi914) [reply src/app/screens/Receive/index.tsx:148]: probably we can add a loading state if we make API call when button is clicked and not when component is rendered. WDYT?

if we add one here. without clicking on button user will see a loading spin. will this be good UX?

File: src/app/screens/Receive/index.tsx (hunk 2 above)
Comment (pavanjoshi914) [reply src/app/screens/Receive/index.tsx]: probably cause pr was created before lnurl-redeem pr was merged!  merged master into the branch, resolved!

**Review by reneaaron — APPROVED** [review]
tACK

**Review by rolznz — APPROVED** [review]

**Pull request merged**